editor/gui/output_dock.cpp

```diff
@@ -45,7 +66,18 @@ void OutputDock::draw(float deltaMilliseconds)
 					ImGui::PushStyleColor(ImGuiCol_Text, (const ImVec4&)EditorSystem::GetSingleton()->getNormalColor());
 				}
 
-				ImGui::Text("%s", outputString.second.c_str());
+				if (outputString.first == "Error" && outputStateMode != (int)OutputState::Warning)
+				{
+					ImGui::Text("%s", outputString.second.c_str());
+				}
+				else if (outputString.first == "Warning" && outputStateMode != (int)OutputState::Error)
+				{
+					ImGui::Text("%s", outputString.second.c_str());
+				}
+				else if (outputStateMode == (int)OutputState::All)
+				{
+					ImGui::Text("%s", outputString.second.c_str());
+				}
```

Comment:
Try to make it so that we can write this.

```cpp

int filter;

if (filter & Filter::Warning)
{
	// Print text in yellow
}

if (filter & Filter::Error)
{
	// Print text in red
}

if (filter & Filter::Print)
{
	// Print text in white
}
```

Reply:
Changed button to checkboxes to implement this.